docs/01-app/01-getting-started/07-caching-and-revalidating.mdx

```diff
@@ -0,0 +1,299 @@
+
+```ts filename="app/lib/data.ts" switcher
+export async function getPosts(slug: string) {
+  'use cache'
+  const data = await fetch(`/api/posts/${slug}`)
+  return data.json()
+}
+```
+
+```js filename="app/lib/data.js" switcher
+export async function getPosts(slug) {
+  'use cache'
+  const data = await fetch(`/api/posts/${slug}`)
+  return data.json()
+}
+```
+
+With the approach above, you can call the function throughout your application code and the same cache entry will be reused as long as the arguments are the same.
+
+Alternatively, you can also cache all functions within a file by adding the `'use cache'` directive at the top of the file:
+
+```ts filename="app/lib/data.ts" switcher
+'use cache'
+
+export async function getPosts() {
+  const data = await fetch(`/api/posts`)
+  return data.json()
+}
+
+export async function getPostBySlug(slug: string) {
+  const data = await fetch(`/api/posts/${slug}`)
+  return data.json()
+}
+```
+
+```js filename="app/lib/data.js" switcher
+'use cache'
+
+export async function getPosts() {
+  const data = await fetch(`/api/posts/`)
+  return data.json()
+}
+
+export async function getPostBySlug(slug: string) {
+  const data = await fetch(`/api/posts/${slug}`)
+  return data.json()
+}
+```
+
+## Revalidating
+
+Revalidation allows you to update cached content without having to rebuild your entire application. It's useful for content that _sometimes_ changes, but would benefit from being cached to improve your application'sperformance.
+
+In Next.js, there are two types of revalidation:
+
+- [Time-based](#time-based-revalidation): Based on a time interval (e.g. every hour).
+- [On-demand](#on-demand-revalidation): Triggered by a specific event (e.g. a CMS webhook).
+
+## Time-based Revalidation
+
+You can use the [`cacheLife` function](/docs/app/api-reference/functions/cacheLife) to define a time interval for how long a cached value should remain stale before it's revalidated.
+
+`cacheLife` comes with [default cache profiles](/docs/app/api-reference/functions/cacheLife#default-cache-profiles) such as `'hour'`, `'day'`, and `'week'`. These profiles can be [customized](/docs/app/api-reference/functions/cacheLife#custom-cache-profiles), if needed.
+
+To use `cacheLife`, import it from `next/cache` and nest it within the **scope** of the `'use cache'` directive and the function. For example, to revalidate the blog page after one hour:
+
+```tsx filename="app/blog/page.tsx" highlight={3,6} switcher
+'use cache'o/>:.
+
+import { cacheLife } from 'next/cache'
+
+export default async function Page() {
+  cacheLife('hour')
+  return <Posts posts={posts} />
+}
+```
+
+```jsx filename="app/blog/page.js" highlight={3,6} switcher
+'use cache'
+
+import { cacheLife } from 'next/cache'
+
+export default async function Page() {
+  cacheLife('hour')
+  return <Posts posts={posts} />
+}
+```
+
```

Comment:
Maybe could stand for a collapsed "deep dive" that explains the stale-while-revalidate semantics, e.g. after one hour it's stale, updates in the background, and then the _next_ request sees the fresh content.

Reply:
FYI I just landed here while searching about how stale-while-revalidate works with the `use cache` directive so it would be good to have it on the docs indeed.